src/boot/pe.c

```diff
@@ -255,6 +283,20 @@ static void pe_locate_sections_internal(
                                         continue;
                         }
 
+                        /* handle efifw section which works very much like .dtbauto */
+                        if (pe_section_name_equal(section_names[i], ".efifw")) {
+                                /* can't match without validate_base */
+                                if (!validate_base)
+                                        break;
+                                if (!pe_use_this_firmware(
+                                                    (const uint8_t *) SIZE_TO_PTR(validate_base) + j->VirtualAddress,
+                                                    j->VirtualSize,
+                                                    device_table,
+                                                    device,
+                                                    (PTR_TO_SIZE(j) - PTR_TO_SIZE(section_table)) / sizeof(*j)))
+                                        continue;
+                        }
+
```

Comment:
this is the exact same code as with the dtbauto handling above, hence merge the two?